enlighten/Controller.py

```diff
@@ -541,24 +543,15 @@ def connect_new(self, other_device=None):
         # try to connect to this device
         ####################################################################
         
-        # The string for a fake usb device used to create a deviceID is always
-        # hashOfSpecName:0x16384:111111:111111
-        # This line checks if the address and bus (last two tok) do not correspond to a Mock device
-        # A BLE device will already have a set device_type of BLEDevice so let it go
-        if isinstance(new_device_id.device_type, BLEDevice):
-            pass
-        elif str(new_device_id.address) != '111111' and str(new_device_id.bus) != '111111':
-            new_device_id.device_type = RealUSBDevice(new_device_id)
-        try:
-            self.bus.device_ids.remove(new_device_id)
-        except:
-            log.error(f"failed to remove new device {new_device_id} from bus devices")
+        self.bus.device_ids.remove(new_device_id)
+        if new_device_id in self.other_devices:
+            self.other_devices.remove(new_device_id)
 
         # attempt to connect the device
         if new_device_id.is_andor():
             self.marquee.info("connecting to XL spectrometer (please wait)", persist=True)
         else:
-            self.marquee.info("connecting to 0x%04x:0x%04x" % (new_device_id.vid, new_device_id.pid), persist=True)
+            self.marquee.info(f"connecting to {new_device_id}", persist=True)
```

Comment:
I should add as it's easy to miss, the marquee will now say <DeviceID {BLE | USB | MOCK} {INFO}>. Not sure if you want maybe the DeviceID and <> stripped.

Reply:
I'm okay with the verbose device description for now.  We can do a "friendliness / simplification" pass later, when all the cool new stuff is fully tested and working...it's reasonable to default to over-informative whilst in R&D.